test/test_autograd.py

```diff
@@ -3717,6 +3717,21 @@ def hook(t_):
         with self.assertRaisesRegex(RuntimeError, "expects the current backward to be executed with multithreading disabled"):
             t.backward()
 
+    def test_view_replay_enabled(self):
+        def f(x):
+            out = x.clone().view(-1)
+            # mutate the view, triggering autograd view-replay logic
+            out.add_(1)
+            return out
+
+        x = torch.ones(2, 2, requires_grad=True)
+        with torch.autograd._force_original_view_tracking(True):
```

Comment:
Could you also test that without this mode, you do get an AsStridedBackward?
I feel like with the current version of this PR that would fail.

Reply:
good point, I'll add that